crates/wasmtime/src/config.rs

```diff
@@ -2165,6 +2159,26 @@ impl Config {
             bail!("wmemcheck (memory checker) was requested but is not enabled in this build");
         }
 
+        #[cfg(not(feature = "component-model"))]
+        if features.component_model() {
+            bail!("component-model support was requested but is not enabled in this build (requires the `component-model` crate feature)");
+        }
+
+        #[cfg(not(feature = "component-model-async"))]
+        if features.component_model_async() {
+            bail!("component-model-async support was requested but is not enabled in this build (requires the `component-model-async` crate feature)");
+        }
+
+        #[cfg(not(feature = "gc"))]
+        if features.gc_types() || features.gc() {
+            bail!("gc support was requested but is not enabled in this build (requires the `gc` crate feature)");
+        }
```

Comment:
For this we should be able to remove this check entirely. I commented a bit above but on this specifically it's where `gc` is  safe to enable and it's just GC-using-things that aren't safe to enable. This is mostly because the GC proposal was larger than *just* GC references, it also introduced many other highly-related but still somewhat orthogonal things. The goal is to use `GC_TYPES` to separate out the runtime requirement but otherwise all the other stuff in the GC proposal is still valid.